src/mapping.ts

```diff
@@ -1,410 +1,477 @@
-import { ethereum, BigInt, Address } from "@graphprotocol/graph-ts";
+import { Address, BigInt, ethereum, log } from "@graphprotocol/graph-ts";
 
 import {
+  OptionType,
+  Claim,
   Account,
   ERC1155Contract,
   ERC1155Transfer,
-  Token,
 } from "../generated/schema";
 
 import {
-  OptionSettlementEngine,
+  OptionSettlementEngine as OSEContract,
   ApprovalForAll as ApprovalForAllEvent,
-  ClaimRedeemed,
-  FeeAccrued,
-  FeeSwept,
-  NewOptionType,
-  OptionsExercised,
-  OptionsWritten,
+  ClaimRedeemed as ClaimRedeemedEvent,
+  FeeAccrued as FeeAccruedEvent,
+  FeeSwept as FeeSweptEvent,
+  NewOptionType as NewOptionTypeEvent,
+  OptionsExercised as OptionsExercisedEvent,
+  OptionsWritten as OptionsWrittenEvent,
   TransferBatch as TransferBatchEvent,
   TransferSingle as TransferSingleEvent,
   URI as URIEvent,
-  FeeToUpdated,
-  FeeSwitchUpdated,
+  FeeToUpdated as FeeToUpdatedEvent,
+  FeeSwitchUpdated as FeeSwitchUpdatedEvent,
 } from "../generated/OptionSettlementEngine/OptionSettlementEngine";
-import { Option, Claim } from "../generated/schema";
-
-import { constants } from "./constants";
-
-import { decimals } from "./decimals";
-
-import { events } from "./events";
-
-import { transactions } from "./transactions";
 
 import { fetchAccount } from "./fetch/account";
-
 import {
   fetchERC1155,
-  fetchERC1155Token,
   fetchERC1155Balance,
-  fetchERC721Operator,
+  fetchERC1155Operator,
+  fetchERC1155Token,
   replaceURI,
 } from "./fetch/erc1155";
-import { exponentToBigDecimal, getTokenPriceUSD } from "./utils/price";
-import { ZERO_ADDRESS } from "./utils/constants";
-import { ERC20 } from "../generated/OptionSettlementEngine/ERC20";
+
 import {
-  initializeToken,
-  loadOrInitializeFeeSwitch,
-  loadOrInitializeToken,
-  updateTokenDayData,
-  updateValoremDayData,
+  fetchOptionSettlementEngine,
+  fetchToken,
+  fetchDailyTokenMetrics,
+  fetchDailyOSEMetrics,
+  handleDailyMetrics,
+  RedeemOrTransferAmounts,
```

Comment:
Should "Redeem" be lower-cased here, to match the others?

Reply:
RedeemOrTransfer amounts is a type/struct, so in this case it should be capitalized.